src/org/dux/stracetool/StraceCall.java

```diff
@@ -9,20 +9,39 @@
     public final String call;
     public final String[] args;
     public final boolean knownReturn;
-    public final int returnValue;
+    public final int returnValue;        // for strace
+    public final String returnMessage;   // for procmon
 
+    // Linux constructor 1
```

Comment:
We should have a factory or builder instead of differentiating this based on which constructor is called.

Reply:
I think this is a very good idea. I'm just not exactly sure how to implement it in a meaningful way. 

I don't think a builder is appropriate because we can't support somebody setting both a return message and return value (since Linux and Windows each only set one and for each I have to auto-initialize the other). Some properties will also only be meaningful on Windows (i.e. all the properties you can enable in Process Monitor).

Regarding a factory, I think a complicated inheritance hierarchy would be overkill, so my current implementation would basically be a wrapper around the constructors -- i.e. three static methods to wrap three constructors. I don't know if that's what you had in mind.

Reply:
I had in mind the second - sorry I was unclear.